server/src/http_login.go

```diff
@@ -131,6 +132,35 @@ func httpLogin(c *gin.Context) {
 				http.Error(w, "That is not the correct password.", http.StatusUnauthorized)
 				return
 			}
+
+			// If they have provided a new password, replace the old one
+			if data.NewPassword != "" {
+				var passwordHash string
+				if v, err := argon2id.CreateHash(data.NewPassword, argon2id.DefaultParams); err != nil {
+					logger.Error("Failed to create a hash from the submitted password for " +
+						"\"" + data.Username + "\": " + err.Error())
+					http.Error(
+						w,
+						http.StatusText(http.StatusInternalServerError),
+						http.StatusInternalServerError,
+					)
+					return
+				} else {
+					passwordHash = v
+				}
+				// Update the user
+				if err := models.Users.ChangePassword(user.ID, passwordHash); err != nil {
+					logger.Error("Failed to update user \"" + data.Username + "\" password: " + err.Error())
+					http.Error(
+						w,
+						http.StatusText(http.StatusInternalServerError),
+						http.StatusInternalServerError,
+					)
+					return
+				} else {
+					logger.Info("Password update for user \"" + data.Username + "\"")
+				}
+			}
 		}
 	} else {
 		// Check to see if any other users have a normalized version of this username
```

Comment:
Should we be erroring here if `data.NewPassword != ""`?

Reply:
My point is that if someone is making a change password request, and the account doesn't exist, it seems apt to have the request fail instead of create a new account with the specified username + current password.

Reply:
That part of the code never runs if this is a new user

Reply:
I know. I'm posting a comment on the `else` for `if exists {`. If `!exists && data.newPassword != ""`, for me that is alarming and we should error. Right now the code will proceed to create a new user as though it wasn't a "change password" request. It's not a huge deal since it seems like this can never happen in the current UI, but I just want to make sure I'm actually understood